pkg/apis/duck/v1/delivery_types.go

```diff
@@ -155,3 +155,52 @@ type DeliveryStatus struct {
 	// +optional
 	DeadLetterSinkURI *apis.URL `json:"deadLetterSinkUri,omitempty"`
 }
+
+func (d *DeliverySpec) Merge(other *DeliverySpec) *DeliverySpec {
+	if d == nil {
```

Comment:
If d is nill it would not have the Merge method (?)

Reply:
Go is special, we can call methods on nil pointers :), see associated unit tests